### ISSUE TITLE
Code highlighting: Use separate color for built-in functions

### ISSUE THREAD

**Issue by Lukas-1** [rstudio/rstudio#6103]
I have recently discovered the option to "Highlight R function calls", and I like it a lot. However, currently, any word with a function-like syntax is highlighted, regardless of whether a function definition exists or not.

It would be useful if it were possible to define a separate color for RStudio themes for known/built-in functions. This way, I would immediately recognize typos, for example, "assing" instead of "assign".

**Comment by kevinushey** [comment]
You can rely on RStudio's diagnostics system for this. In `Tools -> Code -> Diagnostics`, make sure that:

```
[x] Warn if variable used has no definition in scope
```

is checked.

<img width="329" alt="Screen Shot 2020-01-28 at 10 56 04 AM" src="https://user-images.githubusercontent.com/1976582/73295650-ecc1c380-41bc-11ea-8dad-9974238f807d.png">

**Comment by Lukas-1** [comment]
I tried enabling "Warn if variable used has no definition in scope", but I found it more distracting than useful.

The problem is that it will flag variables that I introduce by loading .RData objects, or sourcing other R scripts, so it finds very very many false positives.

I would just like to have a different color for built-in functions, and perhaps also some core functions from commonly used packages.
